### ISSUE TITLE
Builds don't run because of setup.py - issue is happening suddenly

### ISSUE THREAD

**Issue by veenamj** [rtfd/readthedocs.org#3296]
## Details

* Project URL: http://readthedocs.org/projects/qubole-documentation/
* Build URL (if applicable): http://readthedocs.org/projects/qubole-documentation/builds/6302748/
* Read the Docs username (if applicable): 

## Expected Result
Build should pass 

*A description of what you wanted to happen*
Build should run and pass for some time. 
## Actual Result
python setup.py install --force
python: can't open file 'setup.py': [Errno 2] No such file or directory
*A description of what actually happened*

Only for the last 20-24 hours, we are getting the above error that python can't open file setup.py. 
Before the last 1 day, all builds ran fine. We did not do any change in there. I am not sure what the actual cause of this problem is. Can anyone please help us out? Thanks in Advance.

**Comment by xknat** [comment]
i am having same issue.

**Comment by ebertin** [comment]
Same here. Something appears to be broken.

**Comment by krzychb** [comment]
Hi guys,

This is bad. The last three builds ended like that.

* Project URL: https://esp-idf.readthedocs.io/en/latest/?badge=latest#
* Build URL (if applicable): 
  1. https://readthedocs.org/projects/esp-idf/builds/6303064/
  2. https://readthedocs.org/projects/esp-idf/builds/6302622/
  3. https://readthedocs.org/projects/esp-idf/builds/6298978/


![image](https://user-images.githubusercontent.com/12953787/33134748-d38aee18-cfa0-11e7-9470-88107b2855fd.png)

cc: @projectgus

**Comment by safwanrahman** [comment]
Hi,
We have noticed the issue and going to be work on this to fix as soon as possible.

The whole thing is done with volunteering work, so please keep patience.

**Comment by gtalarico** [comment]
Came here to report this. Strange things is, I actually have to RTD pages both pointing to the same Repo, once builds the other stopped building (same settings on both)

Build (failed)
https://readthedocs.org/projects/revitpythonwrapper/builds/6306835/

Build(works)
https://readthedocs.org/projects/rpw/builds/6306827/

![image](https://user-images.githubusercontent.com/9513968/33147519-018fea26-cf96-11e7-8dc8-f129d0476d58.png)

**Comment by humitos** [comment]
@gtalarico one of your build is failing because you have the "Install project" marked

![captura de pantalla_2017-11-22_15-21-37](https://user-images.githubusercontent.com/244656/33148218-fb06aec6-cf98-11e7-945a-97d1a95cc114.png)

and your project doesn't have a `setup.py` file.

In the build that it's passing, you don't have that option marked (the `setup.py install` command is not being executed)

**Comment by humitos** [comment]
I would say that the same is happening to @veenamj and @krzychb. Can you all check this and confirm, please?

**Comment by gtalarico** [comment]
🤦‍♂️
That was it. I must have selected by accident. 
Fixed for me, thanks!

**Comment by veenamj** [comment]
Many thanks @humitos ! The build is working now. I am not sure if all these days that setup.py option was selected or not. So, Im closing this issue.

**Comment by ebertin** [comment]
Thanks, works fine for me too (used to work before with the virtualenv option although there was no setup.py)!

**Comment by agjohnson** [comment]
This was discussed in #3095: https://github.com/rtfd/readthedocs.org/pull/3095#issuecomment-339881383

We don't have a pattern for warning on a particular command failure yet, but ultimately should raise a more helpful error here. Disabling setup.py installation is the correct fix.

**Comment by xknat** [comment]
@humitos this fixed my issue,
i am sure more users will be having same issue as it did work in the past and i did not change any settings at all.

good work tho~!

**Comment by gtalarico** [comment]
That makes sense
But even tough it was silenced, the old builds were going through 
The recent nonnsilent failures were keeping the new builds from going through (docs were not updating)

**Comment by krzychb** [comment]
@safwanrahman, @humitos:

Thank your for checking this issue and clarification.

The build failure is now gone and this is a great news!

Why original builds below were reported as "Passesd" or "Build completed" (green) while actually they failed?

  1. https://readthedocs.org/projects/esp-idf/builds/6303064/
  2. https://readthedocs.org/projects/esp-idf/builds/6302622/
  3. https://readthedocs.org/projects/esp-idf/builds/6298978/

![image](https://user-images.githubusercontent.com/12953787/33166813-ea92f4ac-d03b-11e7-9fff-dbd74dadf960.png)

**Comment by humitos** [comment]
@krzychb that's another issue and should be fixed after #3293 and/or #3292

**Comment by krzychb** [comment]
Sounds good. I am happy to see you have it already in the pipeline. 
Looking forward to see both PR merged!

Will these PRs also cover a case when a build is stuck in "Building" state for several hours? 
https://readthedocs.org/projects/esp-idf/builds/6308479/
https://readthedocs.org/projects/esp-idf/builds/6308480/
https://readthedocs.org/projects/esp-idf/builds/6308949/

![image](https://user-images.githubusercontent.com/12953787/33177427-da7c258a-d062-11e7-8045-57f67f31d799.png)

**Comment by humitos** [comment]
Don't think so.

If something bad happened and the build didn't update its status, I think it will never do it, ha. I'm not sure anyway (maybe there is a task or something that I'm not familiar with).

Those PR won't make that happen again "in theory".

**Comment by Tareque16** [comment]
Where is this advance settings for :
![33148218-fb06aec6-cf98-11e7-945a-97d1a95cc114](https://user-images.githubusercontent.com/16764227/36749770-f02e518c-1c25-11e8-9990-b80472cf41ab.png)

**Comment by stsewd** [comment]
@Tareque16  go to the admin panel of your project, and then to advanced settings.

**Comment by svenevs** [comment]
Hello, I wanted to report that recently I had to create a fake `setup.py` to get past this, **even though** I do **not** have the "Install your project inside a virtualenv using `setup.py install`" checked.

I'm trying to track down an unrelated bug, but the RTD project is here: https://readthedocs.org/projects/sphinx-rtd-tocdepth-confusion-demonstration
